### PR TITLE
Prometheus: 2.30.3 -> 2.35.0

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@o11y.eu>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fpletz — APPROVED** [review]
Thanks and :+1: on the changes, but we shouldn't package a release candidate.

**Comment by K900** [comment]
Maybe mark this as draft until stable 2.35 release, just in case?

**Comment by roidelapluie** [comment]
```
$ nix build  .#prometheus
error: builder for '/nix/store/vd3zil9bbd6zlj51ahzpkbgsfcibw2pf-prometheus-2.35.0-go-modules.drv' failed with exit code 1;
       last 6 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/lqbah4qvswff5vc065lrcidss9k358jm-source
       > source root is source
       > patching sources
       > configuring
       > building
       For full logs, run 'nix log /nix/store/vd3zil9bbd6zlj51ahzpkbgsfcibw2pf-prometheus-2.35.0-go-modules.drv'.
error: 1 dependencies of derivation '/nix/store/w3gf9pvlsixfva4nrg588l8c5kfv3ssf-prometheus-2.35.0.drv' failed to build
```

(I don't know why this does not work)

**Comment by roidelapluie** [comment]
Fixed and ready to go!

**Comment by K900** [comment]
Is 2.35 stable yet?

**Review by K900 — APPROVED** [review]
LGTM, tested on aarch64-linux, both bare and with Grafana.

**Comment by bryanhonof** [comment]
LGTM, tested on x86_64-linux through `gh pr checkout 168706 && nix -vL build .#nixosTests.prometheus`.

**Comment by roidelapluie** [comment]
Can this be merged?

**Comment by Artturin** [comment]
please use `prometheus` (the attribute name) as the commit msg, then ofborg will build the package and tests

**Comment by roidelapluie** [comment]
like this?

File: pkgs/servers/monitoring/prometheus/default.nix [hunk 1]
```diff
@@ -30,131 +28,85 @@
 }:
 
 let
-  version = "2.30.3";
+  version = "2.35.0";
 
   src = fetchFromGitHub {
```
Comment (terinjokes) [inline pkgs/servers/monitoring/prometheus/default.nix:33]: Should be possible to move this into `buildGoModule` as it's no longer needed by other variables.

File: pkgs/servers/monitoring/prometheus/default.nix [hunk 2]
```diff
@@ -30,131 +28,85 @@
...
     rev = "v${version}";
     owner = "prometheus";
     repo = "prometheus";
-    sha256 = "1as6x5bsp7mxa4rp7jhyjlpcvzqm1zngnwvp73rc4rwhz8w8vm3k";
+    sha256 = "sha256-h0uBs3xMKpRH3A1VG5xrhjXVAT7GL5L3UZWb3HJ2Fz4=";
   };
 
   goPackagePath = "github.com/prometheus/prometheus";
```
Comment (terinjokes) [inline pkgs/servers/monitoring/prometheus/default.nix:40]: I don't think this is required anymore.

**Comment by K900** [comment]
Should we try reenabling tests?

**Comment by K900** [comment]
I'd even try enabling Darwin too, it's been almost half a year of new releases, we can always turn it back off if it breaks.

**Comment by roidelapluie** [comment]
> I'd even try enabling Darwin too, it's been almost half a year of new releases, we can always turn it back off if it breaks.

prometheus tests are still broken in darwin.

**Comment by roidelapluie** [comment]
> Is that a known upstream thing?

yes, I remember someone complained a few days ago (without filling a github issue).

**Comment by K900** [comment]
Can also confirm aarch64-linux tests pass on my pretty considerably resource constrained Pi4, so we should be good here.

**Review by mweinelt — APPROVED** [review]
Builds and tests complete for me on aarch64-linux and x86_64-linux.

**Review by Ma27 — APPROVED** [review]

**Pull request merged**